lighthouse-core/test/report/html/renderer/report-renderer-test.js

```diff
@@ -288,9 +288,9 @@ describe('ReportRenderer', () => {
     });
 
     afterAll(() => {
-      global.window = undefined;
-      global.Node = undefined;
-      global.Element = undefined;
+      delete global.window;
```

Comment:
Revert? Unless they're necessary now, seems bad to `delete` things on `global` if we can avoid it (even if normally running inside a `vm.runInContext()`)

Reply:
I'm surprised. How could it be worse? This seems better. No trace of global pollution if you delete.

Reply:
> vm.runInContext()

Jest does this?

Reply:
> I'm surprised. How could it be worse? This seems better. No trace of global pollution if you delete.

`delete` [causes a lot of performance issues](https://stackoverflow.com/a/59432008) (sometimes unexpected), both local and non-local. My general rule for my own code is never use it, like `with` or modifying builtin prototypes.

The global object is different than normal objects and I don't know how things differ with it, and there's an argument to be made for personal preference when there's no benchmarking to base performance decisions on (what's the downside of purging caches and deopting if we're already in a jsdom circus of Proxies?), but for otherwise-unchanged existing code that works fine with `undefined`, it seems better to leave it alone without a reason to change it.

> > vm.runInContext()
> 
> Jest does this?

yeah, that's how it keeps tests isolated from each other, rewrites `require()` without breaking the calling code's `require()`, etc:

https://github.com/facebook/jest/blob/f1ae80554a600cce83d4c3cd54cf6c395745751b/packages/jest-runtime/src/index.ts#L1104-L1109 (probably with a few other places)